File: Pythonleetcode/Leetcode258.py


class Solution(object):
    def addDigits(self, num):
        def digui(n):
            m=0
            while n!=0:
                m=m+(n%10)
                n=int(n/10)
            return m
        while num>=10:
            num=digui(num)
        return num



print(Solution().addDigits(10))

File: CppPractice/Project1/Project1/Leetcode258.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

using namespace std;


class Solution {
public:
    int addDigits(int num) {
        int n;
        int remain;
        int temp_sum;
        while (num>=10)
        {
            n = num;
            temp_sum = 0;
            while (n>=10)
            {
                temp_sum += n % 10;
                n = n / 10;

            }
            temp_sum += n;
            num = temp_sum;
           

        }
        return num;
    }
};